src/gluonts/mx/model/estimator.py

```diff
@@ -117,70 +117,70 @@ def create_predictor(
         """
         raise NotImplementedError
 
+    def create_training_data_loader(
+        self, data: Dataset, **kwargs
+    ) -> DataLoader:
+        raise NotImplementedError
+
+    def create_validation_data_loader(
+        self, data: Dataset, **kwargs
+    ) -> DataLoader:
+        raise NotImplementedError
+
     def train_model(
         self,
         training_data: Dataset,
         validation_data: Optional[Dataset] = None,
         num_workers: Optional[int] = None,
         num_prefetch: Optional[int] = None,
         shuffle_buffer_length: Optional[int] = None,
-        **kwargs,
+        cache_data: bool = False,
     ) -> TrainOutput:
         transformation = self.create_transformation()
 
-        # ensure that the training network is created within the same MXNet
-        # context as the one that will be used during training
-        with self.trainer.ctx:
-            trained_net = self.create_training_network()
-
-        input_names = get_hybrid_forward_input_names(trained_net)
-
-        training_data_loader = TrainDataLoader(
-            dataset=training_data,
-            transform=transformation + SelectFields(input_names),
-            batch_size=self.batch_size,
-            stack_fn=partial(
-                batchify,
-                ctx=self.trainer.ctx,
-                dtype=self.dtype,
-            ),
+        transformed_training_data = TransformedDataset(
```

Comment:
Ahh, now I see. I think I would prefer to have the transformed data chained in the `Estimator` as we had before. This distributes what is happening into too many places, I think.

Reply:
The point is, you may want to exploit the difference between completely deterministic transformations (i.e. data pre-processing) from random ones. For example, you may want to cache the former (like it is done here) to speed up training.

I think in general these are two very different kinds of transformations: one part says how all data entries in the datasets should be adjusted to fit the model's needs, the other instructs the data loader on how to split instances that need to be stacked into batches.

But I'll reconsider all of this, I'm thinking of reducing the scope of this PR a bit to make it less breaking